### PR TITLE
Add store locator demo

### PR DESCRIPTION
**Pull request by AndrewSepic**
>[!NOTE]
> ## Update
> Additional Changes to this repo outside of the store-locator functionality
> * Importing the accessToken from `demo-components` across the app.
> * Adding the `demo-components` folder to the yarn workspace to allow the demo-store-locator to include imported component in the build
> * Add the `demo-components` filepath to the `content` value of tailwind.config.js to allow tailwind to process styles on the imported components.  See [here](https://github.com/tailwindlabs/tailwindcss/discussions/7720) for more info.

This PR introduces a new Store Locator Demo powered by Vite.js/React. 

The store locator demo was based loosely on the real estate demo and modified the UI & UX patterns to better suit the purpose.  

## Data
The data for this project comes from a custom map style  & tileset created under the [docs `examples` account](https://studio.mapbox.com/styles/examples/cm0foo08s01tn01qq2dzccr6i/edit/#10.61/35.3763/-119.0334).  This tileset data has been processed with a tileset recipe to utilize MTS clustering ([see tileset here](https://studio.mapbox.com/tilesets/examples.store-locations-clustering/#8.26/40.66/-73.776)), creating a tileset layer of locations which are clustered ( for low zoom levels) and which are not clustered (for high zooms).  The data is a subset of Starbucks locations from this [repo](https://github.com/chrismeller/StarbucksLocations) cross referenced with active Starbucks which came from our dataplor.

## Mobile vs Desktop
This demo modifies UX/Layout based on small screens vs large.  Large screens use the left sidebar to display a listing, which allows you to activate a marker on the map for an active Location, while on mobile, the listing is hidden and store information is loaded into a pop up when you interact with the store-locations layer. 

## User Location
The application will attempt to get the user's location through the browsers `navigator.geolocation`. If the user shares their location it will fly to their location & generate a map pulse animation on the user. After that point, the user can always click the 'use my location' button to fly back to their location.  

If a user denies the access to their location, the app will fly to Seattle as a demo city. 

## SearchBox
Searchbox is scoped to return only US search results. When a new search query is made, previously loaded results are removed. And the search input notes whether the user location is being used as the map location.

## Locations Listing
The Location Listing component renders on Desktop screens.  It will be empty on Zoom levels of 10 or higher displaying 'Zoomed too far out' as a message.  It will scroll infitintely to load many locations, and interacting with locations on the map, or in the listing will use the `.scrollIntoView()` web element API to move the active listing into view in the sidebar.

## Design & Graphics
The design & graphics Logo by me are based on a fictional coffee shop chain with free map pun care of Bill Morris.

## Testing 
This current branch is deployed for testing at https://labs.tilestream.net/demo-store-locator/

### CODE REVIEW

File: projects/demo-store-locator/src/Context/AppContext.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export const AppContext = createContext();
+
+// Create a provider component
+export const AppContextProvider = ({ children }) => {
+
+  // Stores the feature that the user is currently viewing (triggers the marker/popup)
+  const [activeFeature, setActiveFeature] = useState()
+  // activeLocation rendered on the map
+  const [ activeLocation, setActiveLocation ] = useState(null);
+  // The data returned from the visible viewport of the map. Rendered in LocationListing
```
Comment (chriswhong) [inline projects/demo-store-locator/src/Context/AppContext.jsx:13]:
```suggestion
  // The store data in the current map viewport. Rendered in LocationListing, updated whenever the viewport changes
```

File: projects/demo-store-locator/src/LocationListing.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// The LocationListing component renders GeoJSON feature information retrieved by
```
Comment (chriswhong) [inline projects/demo-store-locator/src/LocationListing.jsx:1]: Can we rename this to `CardList`?

File: projects/demo-store-locator/src/Map/index.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  // This demo imports accessToken from  `demo-components/accessToken`, to use this project
+  // for your purposes, replace `accessToken` below with your own Access Token available
+  // at (https://account.mapbox.com/) - This token is also used in SearchBoxWrapper
+
+  mapboxgl.accessToken = accessToken;
+
+  useEffect(() => {
+    const map = (mapRef.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: 'mapbox://styles/examples/cm0foo08s01tn01qq2dzccr6i',
```
Comment (chriswhong) [inline projects/demo-store-locator/src/Map/index.jsx:44]:
```suggestion
      // this is a custom style, created with store location data loaded via Mapbox Tiling Service and styled in Mapbox Studio.
      style: 'mapbox://styles/examples/cm0foo08s01tn01qq2dzccr6i', 
```

File: projects/demo-store-locator/src/Map/index.jsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      setMapLoaded(true)
+    })
+
+    // Set the max bounds of the map to the extent of your dataset
+    map.setMaxBounds([
+      [-164.00944, 24.83458], // SouthWest coordinates
+      [-68.52300, 70.17738] // Northeast coordinates
+    ]);
+
+    // Change the cursor to a pointer when the mouse is over a feature in the layer.
```
Comment (chriswhong) [inline projects/demo-store-locator/src/Map/index.jsx:65]:
```suggestion
    // Change the cursor to a pointer when the mouse is over a feature in the store-locations layer.  
```

File: projects/demo-store-locator/src/layout.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types'
+import { Inter } from 'next/font/google'
+import './globals.css'
+
+const inter = Inter({ subsets: ['latin'] })
+
+export const metadata = {
+  title: 'Mapbox Real Estate App Demo',
+  description: 'Generated by create next app'
```
Comment (chriswhong) [inline projects/demo-store-locator/src/layout.jsx:9]:
```suggestion
  description: 'A demo store locator app showcasing various Mapbox products and techniques.'
```

**Review by chriswhong — CHANGES_REQUESTED** [review]
I've left a lot of suggestions to clarify comments, and some requests to rename components and split them into separate files.

Can we also run `prettier` on this project? (I would like to have prettier run on commit in this repo, but we can handle that later)

I have some other requests for the UI/UX which I'll communicate in a different channel.

File: projects/demo-components/package.json [hunk 6]
```diff
@@ -24,6 +24,8 @@
     "@fortawesome/react-fontawesome": "^0.2.0",
     "@mapbox/mapbox-gl-geocoder": "^5.0.1",
     "@mapbox/web-analytics": "^0.9.0",
-    "mapbox-gl": "^2.14.1"
+    "mapbox-gl": "^2.14.1",
+    "react-markdown": "^9.0.1",
```
[the remark this answers is not in the document]
Comment (AndrewSepic) [reply projects/demo-components/package.json:28]: I've removed these from 'demo-store-locator'.  To remove them from `demo-real-estate`, that project needs to get some minor updates to remove their components, import the new components and adapt the display to be consistent with the toggleable layout.  I'd like to PR that in separately, after this is merged.

File: projects/demo-store-locator/src/LocationListing.jsx (hunk 2 above)
Comment (AndrewSepic) [reply projects/demo-store-locator/src/LocationListing.jsx:1]: Sure.

**Review by chriswhong — APPROVED** [review]
Looks great! Glad to see this get across the finish line!

**Pull request merged**